benchmarks/nbody/benchmark.rb

```diff
@@ -82,6 +82,10 @@ def offset_momentum(bodies)
   b.vz = - pz / SOLAR_MASS
 end
 
+# Pre-allocate a planet so that subsequent Planet objects
+# will have the same shape.
+Planet.new(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0)
```

Comment:
From my knowledge of maps/Shapes, this sounds like a hack and a bug of the Shape transition implementation.

There is a set of `Planet.new` calls below, all with the same arguments, all setting the same ivars.
If they don't have the same Shape, it's a bug of the Shape system.